### PR TITLE
MRG: Add head position plotting

### PR DESCRIPTION
**Pull request by Eric89GXL**
~~WIP because I need to look up the correct notation for quaternions...~~

Ready for review/merge.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Okay ready for review/merge from my end

File: mne/viz/_3d.py [hunk 1]
```diff
@@ -52,6 +54,82 @@ def _fiducial_coords(points, coord_frame=None):
...
+    from ..chpi import head_pos_to_trans_rot_t
+    import matplotlib.pyplot as plt
+    from matplotlib.colors import Normalize
+    from mpl_toolkits.mplot3d.art3d import Line3DCollection
+    from mpl_toolkits.mplot3d import axes3d  # noqa: F401
+    if not isinstance(mode, string_types) or mode not in ('traces', 'field'):
+        raise ValueError('mode must be "traces" or "field", got %s' % (mode,))
+    trans, rot, t = head_pos_to_trans_rot_t(pos)  # also ensures pos is okay
+    # trans, rot, and t are for dev_head_t, but what we really want
+    # ar head_dev_t (i.e., where the head origin is in device coords)
```
Comment (jaeilepp) [inline mne/viz/_3d.py]: ar -> are

**Comment by Eric89GXL** [comment]
`[ci skip]` was because that particular commit shouldn't have affected the tests (i.e., would neither break new things nor fix errors). It was done to save CI cycles so other PRs could get faster feedback

**Comment by jaeilepp** [comment]
Does [this](https://travis-ci.org/mne-tools/mne-python/jobs/206837041#L2854) still fail?

**Comment by Eric89GXL** [comment]
Yes the failure is legitimate, and related to `viridis` not being available in old MPL. Before I decide what to do with that, are you (and @agramfort) happy with `viridis` as the color scheme [here](https://3926-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/auto_examples/preprocessing/plot_head_position.html)? If you'd prefer a different one, that might fix the error. Otherwise I can use a fallback to something else for old MPL.

**Comment by Eric89GXL** [comment]
Little fix to the plot (ensure `tail` mode):

https://3930-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/auto_examples/preprocessing/plot_head_positions.html

**Comment by jaeilepp** [comment]
In this case I suppose 'jet' would be a viable option too, but I have no preference. Noticed that the plotter lacks ``show`` and ``block`` options.

**Comment by Eric89GXL** [comment]
Most of our plotters only have `show`, do we want to start adding `block` everywhere?

**Comment by Eric89GXL** [comment]
... and I am [morally opposed to](https://jakevdp.github.io/blog/2014/10/16/how-bad-is-your-colormap/) `jet`, so we can't use that :)

**Comment by agramfort** [comment]
block was necessary when interaction was changing the object. Otherwise
it's not needed.

ok for viridis.

**Comment by Eric89GXL** [comment]
Latest commit should fix problems with old mpl

FYI @agramfort it looks like mpl3d was introduced in 1.4, which came out 2.5 years ago (august '14). We probably don't need to change our version requirements yet, but eventually we might want to

File: mne/viz/_3d.py [hunk 2]
```diff
@@ -52,6 +54,97 @@ def _fiducial_coords(points, coord_frame=None):
...
+        lc.set_array(t[:-1])
+        ax.add_collection(lc)
+        # now plot the head directions as a quiver
+        dir_idx = dict(x=0, y=1, z=2)
+        kwargs = dict()
+        for d, length in zip(direction, [1., 0.5, 0.25]):
+            use_dir = use_rot[:, :, dir_idx[d]]
+            # draws stems, then heads
+            array = np.concatenate((t, np.repeat(t, 2)))
+            if check_version('matplotlib', '1.4'):
```
Comment (jaeilepp) [inline mne/viz/_3d.py]: This is already checked on line 112.

File: mne/viz/_3d.py [hunk 3]
```diff
@@ -52,6 +54,97 @@ def _fiducial_coords(points, coord_frame=None):
...
+        raise ValueError('mode must be "traces" or "field", got %s' % (mode,))
+    trans, rot, t = head_pos_to_trans_rot_t(pos)  # also ensures pos is okay
+    # trans, rot, and t are for dev_head_t, but what we really want
+    # is head_dev_t (i.e., where the head origin is in device coords)
+    use_trans = np.einsum('ijk,ik->ij', rot[:, :3, :3].transpose([0, 2, 1]),
+                          -trans) * 1000
+    use_rot = rot.transpose([0, 2, 1])
+    use_quats = -pos[:, 1:4]  # inverse (like doing rot.T)
+    if isinstance(cmap, string_types) and cmap == 'viridis' and \
+            not check_version('matplotlib', '1.5'):
```
Comment (jaeilepp) [inline mne/viz/_3d.py]: ``if cmap == 'viridis' and not check_version('matplotlib', '1.5'):`` should be enough.

Comment (Eric89GXL) [inline mne/viz/_3d.py]: I wasn't sure it would behave with all varieties of `cmap` arguments that `matplotlib` supports, e.g. `Colormap` objects. Will it?

Comment (jaeilepp) [inline mne/viz/_3d.py]: Yes, string check not needed.

**Comment by Eric89GXL** [comment]
Thanks @jaeilepp comments addressed.

@kambysese feel free to try this if you want

**Comment by agramfort** [comment]
@Eric89GXL please update what's new and merge

**Comment by Eric89GXL** [comment]
Thanks for the reviews

**Pull request merged**